File: 顺序刷/二叉树/src/TreeNode.java
import java.util.Deque;
import java.util.LinkedList;
import java.util.List;

/**
 * 二叉树
 */
public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode next;

    TreeNode() {}
    TreeNode(int val) {
        this.val = val;
    }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }

    /**
     * 根据数组创建二叉树
     * @param list
     * @return
     */
    static TreeNode createTreeFromList(Integer[] list) {
        if (list.length == 0 || list[0] == null) {
            return null;
        }
        Deque<TreeNode> deque = new LinkedList<>();
        TreeNode h = new TreeNode(list[0]);
        TreeNode p = null;
        deque.offer(h);
        int i = 1;
        while(!deque.isEmpty()) {
            p = deque.poll();
            if (i >= list.length || list[i] == null) {
                p.left = null;
            } else {
                p.left = new TreeNode(list[i]);
                deque.offer(p.left);
            }
            ++i;
            if (i >= list.length || list[i] == null) {
                p.right = null;
            } else {
                p.right = new TreeNode(list[i]);
                deque.offer(p.right);
            }
            ++i;
        }
        return h;
    }

    /**
     * 层次打印 二叉树
     * @param root
     */
    static void printTree(TreeNode root) {
        if (root == null) return;
        List<List<Integer>> ans = new LinkedList<>();
        Deque<TreeNode> deque = new LinkedList<>();
        deque.offer(root);
        while(!deque.isEmpty()) {
            int len = deque.size();
            List<Integer> temp = new LinkedList<>();
            for (int i = 0; i < len; i++) {
                root = deque.poll();
                temp.add(root.val);
                if (root.left != null) deque.offer(root.left);
                if (root.right != null) deque.offer(root.right);
            }
            ans.add(temp);
        }
        System.out.println(ans);
    }
}

File: python/BiTree/treeNode.py
class TreeNode:
    def __init__(self, value: int=None) -> None:
        self.value = value
        self.left: TreeNode = None
        self.right: TreeNode = None

    def insert(self, value: int) -> None:
        if value < self.value:
            if self.left is None:
                self.left = TreeNode(value)
            else:
                self.left.insert(value)
        elif value > self.value:
            if self.right is None:
                self.right = TreeNode(value)
            else:
                self.right.insert(value)
    def insert_b(self, value: int) -> None:
        p = self
        while p is not None:
            if value < p.value:
                if p.left is None:
                    p.left = TreeNode(value)
                    break
                else:
                    p = p.left
            elif value > p.value:
                if p.right is None:
                    p.right = TreeNode(value)
                    break
                else:
                    p = p.right
            

    def insert_list(self, values: list) -> None:
        for value in values:
            self.insert(value)
